agent/agent.go

```diff
@@ -113,6 +113,11 @@ func (r *Agent) Run(recipe recipe.Recipe) (run Run) {
 		stream      = newStream()
 		recordCount = 0
 	)
+
+	defer func() {
+		r.runRecordRun(run, getDuration, recipe.Name)
+	}()
```

Comment:
Wouldn't it be better written like this?
```suggestion
	defer r.runRecordRun(run, getDuration, recipe.Name)
```

Reply:
> Wouldn't it be better written like this?

The parameter values in the runRecordRun change their value, so in order to pass the latest value, I used this closure approach.